### PR TITLE
`pdnsutil`: Add `backend-lookup` subcommand

### PR DESCRIPTION
**Pull request by fredmorcos**
### Short description
This can be used to lookup records from a backend using some filtering like type and client subnet.

### Checklist
I have:
- [X] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [X] compiled this code
- [X] tested this code
- [ ] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)

### CODE REVIEW

**Review by Habbie — COMMENTED** [review]
few comments/requests. I'll likely push commits fixing some of them

let me know if that "filtering" word had some meaning to you, I might be missing something

File: docs/manpages/pdnsutil.1.rst [hunk 1]
```diff
@@ -283,6 +283,8 @@ backend-cmd *BACKEND* *CMD* [*CMD...*]
     Send a text command to a backend for execution. GSQL backends will
     take SQL commands, other backends may take different things. Be
     careful!
+backend-lookup *BACKEND* *NAME* *TYPE* *CLIENT-IP-SUBNET*
```
[the remark this answers is not in the document]
Comment (Habbie) [reply docs/manpages/pdnsutil.1.rst]: pushed

File: pdns/pdnsutil.cc [hunk 2]
```diff
@@ -4225,6 +4229,49 @@ try
...
+    DNSPacket queryPacket(true);
+    Netmask clientNetmask(cmds.at(4));
+    queryPacket.setRealRemote(clientNetmask);
+
+    matchingBackend->lookup(type, name, -1, &queryPacket);
+
+    bool found = false;
+    DNSZoneRecord resultZoneRecord;
+    while (matchingBackend->get(resultZoneRecord)) {
+      cout << resultZoneRecord.dr.d_name.toString() << "\t" << resultZoneRecord.dr.getContent()->getZoneRepresentation() << "\t" << std::to_string(resultZoneRecord.dr.d_ttl) << "\t" << QClass(resultZoneRecord.dr.d_class).toString() << "\t" << DNSRecordContent::NumberToType(resultZoneRecord.dr.d_type, resultZoneRecord.dr.d_class) << "\t" << endl;
```
[the remark this answers is not in the document]
Comment (Habbie) [reply pdns/pdnsutil.cc]: pushed

**Comment by Habbie** [comment]
I pushed a few enhancements. Feel free to reorder, squash, edit how you see fit. Nice work, by the way!

**Comment by fredmorcos** [comment]
> let me know if that "filtering" word had some meaning to you, I might be missing something

Nothing other than we're querying records based on some criteria.

**Comment by Habbie** [comment]
the "Update argument name" reminds me that I missed that line when making the arguments optional

**Comment by Habbie** [comment]
> Nothing other than we're querying records based on some criteria.

right, i think removing the word is clearer then

**Comment by fredmorcos** [comment]
> the "Update argument name" reminds me that I missed that line when making the arguments optional

Fixed

**Review by Habbie — APPROVED** [review]

**Pull request merged**